source/distributed/shared_tria.cc

```diff
@@ -134,6 +135,20 @@ namespace parallel
       this->update_number_cache ();
     }
 
+    template <int dim, int spacedim>
+    void
+    Triangulation<dim, spacedim>::
+    copy_triangulation (const dealii::Triangulation<dim, spacedim> &old_tria)
+    {
+      Assert ((dynamic_cast<const dealii::parallel::distributed::Triangulation<dim,spacedim> *>(&old_tria) == NULL),
+              ExcMessage("Cannot use this function on parallel::distributed::Triangulation."));
```

Comment:
I second @tjhei 's concerns. The point is that you do Assert to make sure the function is used with the classes that are tested. If there are other triangulations some day, they won't be automatically tested with this function. What is tested and promised to work is `::Tria` and `p::s::Tria`, until someday there are more tests which confirm correct behaviour with other classes.